scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/TarWriter.java

```diff
@@ -0,0 +1,54 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+
+public class TarWriter {
+
+    private FileOutputStream out;
+    private TarArchiveOutputStream tOut;    
+    private int len;
+    
+    public TarWriter(String outPath) throws IOException{        
+        out = new FileOutputStream(outPath + ".tar");        
+        tOut = new TarArchiveOutputStream(out);
+        tOut.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
+        len = outPath.length();
+    }
+
+    private void writeEntry(File file) throws IOException{
+        Path sourcePath = FileSystems.getDefault().getPath(file.getPath());
+        byte[] sourceBytes = Files.readAllBytes(sourcePath);
+
+        String alteredPath = file.getPath();
+
+        alteredPath = alteredPath.substring(len);
+        TarArchiveEntry tarEntry = new TarArchiveEntry(file, alteredPath);
+        
+        tarEntry.setSize(sourceBytes.length);
+       
+        tOut.putArchiveEntry(tarEntry);
+        tOut.write(sourceBytes);
+        tOut.closeArchiveEntry();
+    }
+    
+    public void writeDir(File rootFile) throws IOException{    
+        File[] fileTree = rootFile.listFiles();
+        if(rootFile.getName().equals(".git")){return;}
```

Comment:
the files/folders should be a configurable list

Reply:
I'm not really sure what you ment by this comment? Could you please expand on this statement.

Reply:
I mean that `.git` will probably not stay the only folder/file that should be excluded, so this should not be hard coded but a configuration property to list all files/folder that should not be compressed here. Does this make more sense?